include/dts-functions.sh

```diff
@@ -1730,72 +1624,90 @@ send_dts_logs(){
 }
 
 check_if_fused() {
-  file_path="/sys/class/mei/mei0/fw_status"
-
-  if [[ ! -f $file_path ]]; then
-    echo "File not found: $file_path"
+  local _file_path
+  _file_path="/sys/class/mei/mei0/fw_status"
+  local _file_content
+  local _hfsts6_value
+  local _line_number
+  local _hfsts6_binary
+  local _binary_length
+  local _padding
+  local _zeros
+  local _bit_30_value
+
+  if ! $FSREAD_TOOL test -f "$_file_path"; then
+    print_error "File not found: $_file_path"
     return 2
   fi
 
-  hfsts6_value=""
-  line_number=1
+  _file_content="$($FSREAD_TOOL cat $_file_path)"
+
+  _fsts6_value=""
+  _line_number=1
   while IFS= read -r line; do
-    if [[ $line_number -eq 6 ]]; then
-      hfsts6_value=$line
+    if [[ $_line_number -eq 6 ]]; then
+      _hfsts6_value="$line"
       break
     fi
-    ((line_number++))
-  done <"$file_path"
+    ((_line_number++))
+  done <<< "$_file_content"
```

Comment:
Could simplify whole loop to one command:

```sh
_hfsts6_value=$(sed '6q;d' <<< "$_file_content")
```

Reply:
Simplifying not always mean making it shorter, sometimes it also means to make it more clear. And I do not agree, IMHO the `sed '6q;d'` is less clear than a loop.

Reply:
Depends on person. What about `_hfsts6_value=$(echo "$_file_content" | head -6 | tail -1)`. No need to remember what `while IFS= read` does.